src/coreComponents/dataRepository/Group.cpp

```diff
@@ -254,6 +259,171 @@ void Group::printDataHierarchy( integer const indent )
   }
 }
 
+void Group::printMemoryAllocation( integer const indent, real64 const threshold )
```

Comment:
I think this is a recursive function starting from group `this` and continuing into the `m_subGroups`.
Could it be possible to make it a free function in the `helpers`, giving the `root` group as an argument?
Like maybe `printDataHierarchy` and `dumpInputOptions` as well. That way `Group` would not carry to many "debugging" functions.

Reply:
yes.

Reply:
I am adding a `Utility's.hpp/cpp` to the `dataRepsitory` component. Naming and further namespace scoping is up for debate.